Display error when image exceeds `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` limit

azrikahar opened the pull request:
Fixes #7996

## Reported Bug

When an image exceeds the configured `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` env variable limit, the image src will break. However we can still view it by clicking `zoom`:

![qwNo9Fv0pa](https://user-images.githubusercontent.com/42867097/133563878-c470ecc8-d74e-46d3-b4c2-3c255e86340b.gif)

## Solution

call the asset api and catch the error if the code is `ILLEGAL_ASSET_TRANSFORMATION`, then diplay an info icon with a message.

Update:

- moved error handling to `@error` event for <img>
- added "generic" error type to show image cannot be loaded, in case the error isn't `ILLEGAL_ASSET_TRANSFORMATION`. (not sure if this will ever happen?)
- added relevant translation keys.

## After Fix

![yldCawqCIv](https://user-images.githubusercontent.com/42867097/133563843-0fbb9f55-b2ae-4bde-9520-ee3227b8fdad.gif)

### Updated look:

- When error code is `ILLEGAL_ASSET_TRANSFORMATION`:

    ![chrome_kIh3yM9sk0](https://user-images.githubusercontent.com/42867097/133722576-cc477503-c3d8-4fb6-bb2b-a64b523f9726.png)

- When error code is generic:

    ![chrome_JlABTkzaME](https://user-images.githubusercontent.com/42867097/133722641-29cfa127-2915-45ce-9f98-d4fcafe6f421.png)

## ~Uncertainty~

~calling the assets API and trigger it in the same try/catch might feel hacky.. but in a weird way it sort of makes sense? still likely that I'm reaching for a quick fix too soon, so do let me know if there's a more appropriate approach.~

~the error message is also not set in stone. I just referred it from the existing error message from the api:~

https://github.com/directus/directus/blob/9ba43c19f8c7ae4add0394ace7f90cbc524177e7/api/src/services/assets.ts#L95-L97

azrikahar committed:
display error when image exceeds transform limit

benhaynes commented:
I defer to @rijkvanzanten on implementation, but for style:

Let's shorten this text and give it a max-width (eg: 300px) so it feels more contained. Potential text update below:

```
Image source too large to preview.
```

josias-r commented:
Unfortunately, I‘m not at the computer to check this myself. But did you also think of the image in the list view of a collection? It was also erroring out

azrikahar commented:
> Unfortunately, I‘m not at the computer to check this myself. But did you also think of the image in the list view of a collection? It was also erroring out

Ya I also noticed that as I was investigating this issue. but I felt like the image in list view + image in files preview are more similar in nature (based on their logic in the source code), so I'm planning to fix those in a separate PR. 👍

azrikahar committed:
use img error event handler

azrikahar commented:
Update (see main post for screenshots):

- moved error handling to `@error` event for <img>
- added "generic" error type to show image cannot be loaded, in case the error isn't `ILLEGAL_ASSET_TRANSFORMATION`. (not sure if this will ever happen?)
- added relevant translation keys.

rijkvanzanten commented:
> added "generic" error type to show image cannot be loaded, in case the error isn't ILLEGAL_ASSET_TRANSFORMATION. (not sure if this will ever happen?)

Smart! We don't expect it to, but anything can happen when it comes to computers and networks, so better safe than sorry 🙂

rijkvanzanten commented on app/src/interfaces/file-image/file-image.vue:
```diff
@@ -7,7 +7,17 @@
+				<span class="message">
+					{{
+						imageError === 'illegalAssetTransformation'
+							? t('image_errors.too_large_to_preview')
```
We already have an "ERRORS" key under translations. Maybe we can simply use the "translateAPIError" utility for this purpose :thinking:

rijkvanzanten commented on app/src/lang/translations/en-US.yaml:
```diff
@@ -561,6 +561,9 @@ upload_files_success: '{count} Files Uploaded'
 click_to_browse: Click to Browse
+image_errors:
+  generic: Cannot load image.
+  too_large_to_preview: Image source too large to preview.
```
Lets move this to the more generic "errors" key

rijkvanzanten committed:
Use generic error translations

rijkvanzanten committed:
Merge branch 'main' into pr/azrikahar/8069

rijkvanzanten committed:
Fix unknown translation check
